pmultiqc/modules/quantms/histogram.py

```diff
@@ -0,0 +1,119 @@
+from collections import OrderedDict, defaultdict
+
+class Histogram:
+
+    def classify(self, plot_cat, with_remains = False, data_cats = None):
+        # with_remains -> Whether there are any remaining values outside the statistical range.
+        # data_cats -> Make statistics and display according to different data categories.
+        # plot_cat = 'frequency' -> Count the number of occurrences of different values.
+        # plot_cat = 'range' -> Count the frequency with which a value occurs over several consecutive ranges.
+        # plot_cat = 'record' -> Assigns values in order according to the different ranges provided.
+
+        self.data = dict()
+        self.data_cats = data_cats
+        if plot_cat == 'frequency':
+            if with_remains:
+                self.plot_cat = 2
+            else:
+                self.plot_cat = 1
+                self.frequency = dict()
+        elif plot_cat == 'range':
+            self.plot_cat = 3
+        elif plot_cat == 'record':
```

Comment:
I am not sure if we really need cat 4.
Do you mean histograms for
E.g. 1, 5, 8 ?

         ^^^ this is not really a histogram.

While 1,2,3,4+ is an application for it.